modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -405,4 +482,26 @@ static void verifyQuery(QueryBuilder queryBuilder) {
         }
     }
 
+    static byte[] encodeRange(String rangeFieldName, byte[] minEncoded, byte[] maxEncoded) {
+        assert minEncoded.length == maxEncoded.length;
+
+        byte[] bytes = new byte[BinaryRange.BYTES * 2];
+        int offset = BinaryRange.BYTES - minEncoded.length;
+        if (offset >= 8) {
```

Comment:
I think we need to keep it? Otherwise we would compute a hash and then not use it? Also the copying of encoded hashes would fail.